### ISSUE TITLE
kubectl get nodes connection refused

### ISSUE THREAD

**Issue by urgent** [coreos/coreos-kubernetes#233]
I updated my boxes and started to get this error when running kubectl from host:

    error: couldn't read version from server: Get https://172.17.4.101/api: dial tcp 172.17.4.101:443: connection refused

I ssh into c1 and open port 443 in iptables and now I can connect:
    `vagrant ssh c1`
    `sudo iptables -A INPUT -m state --state NEW -m tcp -p tcp --dport 443`

**Comment by aaronlevy** [comment]
Hi @urgent, can you provide a little more detail on what you mean by "updated my boxes"? Or possible steps to reproduce the issue?

**Comment by scotty-c** [comment]
Hi @aaronlevy 
I got the same issue when building for the first time.
I followed the instructions https://coreos.com/kubernetes/docs/latest/kubernetes-on-vagrant.html#start-the-machines. To fix the issue I used the above method to add the iptables rule on c1

**Comment by gembin** [comment]
I followed the same instructions https://coreos.com/kubernetes/docs/latest/kubernetes-on-vagrant.html, and didn't change any configuration(all defaults).  And got the same issue, and add the iptables rule still doesn't work.  Using OSX 10.9.5 + VirtualBox 5.0.14

**Comment by urgent** [comment]
Hi @aaronlevy , I just ran `vagrant box update` from inside coreos-kubernetes/multi-node/vagrant/

**Comment by philips** [comment]
Can someone give me the output of `iptable -L` and `route -n` and `ip addr` from hosts exhibiting this. I am super confused why this is happening and how #267 fixes it.

**Comment by hdanes** [comment]
I'm not sure if it's related to this issue; but when I use the stable channel `kubelet` seems te be missing in the vagrant box. Which causes the same error as mentioned above.

$ vagrant ssh c1
CoreOS stable (835.13.0)
Update Strategy: No Reboots
Failed Units: 1
  coreos-cloudinit-309581450.service

$ journalctl -f
Feb 21 12:45:12 c1 systemd[884]: kubelet.service: Failed at step EXEC spawning /usr/bin/kubelet: No such file or directory

In the alpha channel everything is running as expected.

**Comment by robszumski** [comment]
@hdanes That issue is unrelated to this. There is a WIP PR to update the docs to use the new kubelet wrapper: https://github.com/coreos/coreos-kubernetes/pull/276

**Comment by gangefors** [comment]
I seem to have a related problem. It ends up exactly the same as this but it have to do with me being behind a corporate proxy.

Don't know If I should file a new ticket on updating the documentation on how to get this working when behind a proxy or continue on this one?

Anyway, my problem is that the kubelet-wrapper can't contact the Internet since it's not configured correctly for my proxy.

I have this in my log:
```
$ journalctl -f
Mar 16 17:39:54 c1 kubelet-wrapper[2721]: run: Get https://quay.io?ac-discovery=1: dial tcp 54.235.93.246:443: i/o timeout
Mar 16 17:39:57 c1 systemd-timesyncd[648]: Timed out waiting for reply from 193.228.143.13:123 (3.coreos.pool.ntp.org).
Mar 16 17:40:04 c1 systemd[1]: kubelet.service: Service hold-off time over, scheduling restart.
Mar 16 17:40:04 c1 systemd[1]: Stopped kubelet.service.
Mar 16 17:40:04 c1 systemd[1]: Starting kubelet.service...
Mar 16 17:40:04 c1 systemd[1]: Started kubelet.service.
Mar 16 17:40:05 c1 kubelet-wrapper[2750]: image: using image from file /usr/share/rkt/stage1-fly.aci
Mar 16 17:40:05 c1 kubelet-wrapper[2750]: image: searching for app image quay.io/coreos/hyperkube
Mar 16 17:40:06 c1 locksmithd[674]: [etcd.service etcd2.service] are inactive
Mar 16 17:40:06 c1 locksmithd[674]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 5m0s.
Mar 16 17:40:07 c1 systemd-timesyncd[648]: Timed out waiting for reply from 94.247.173.154:123 (3.coreos.pool.ntp.org).
Mar 16 17:40:17 c1 systemd-timesyncd[648]: Timed out waiting for reply from 195.178.181.98:123 (3.coreos.pool.ntp.org).
Mar 16 17:40:20 c1 systemd[1]: kubelet.service: Main process exited, code=exited, status=1/FAILURE
Mar 16 17:40:20 c1 systemd[1]: kubelet.service: Unit entered failed state.
Mar 16 17:40:20 c1 systemd[1]: kubelet.service: Failed with result 'exit-code'.
```

Adding proxy settings with the vagrant-proxyconf plugin doesn't seem to help for the kubelet-wrapper but works for curl and docker.
```
$ docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
03f4658f8b78: Pull complete 
a3ed95caeb02: Pull complete 
Digest: sha256:8be990ef2aeb16dbcb9271ddfe2610fa6658d13f6dfb8bc72074cc1ca36966a7
Status: Downloaded newer image for hello-world:latest

Hello from Docker.
[...]
```

I'm at a loss on how I may continue. I can't find any documentation that describes how I set this up behind a corporate proxy.
If I can get all services running under systemd to be configured with my proxy I think I should be fine but where do I set a systemd global proxy?

**Comment by gangefors** [comment]
I managed to solve my proxy problem for the kubelet-wrapper in the controller and worker by updating the install scripts available in the ./generic folder.

```
Environment=KUBELET_VERSION=${K8S_VER}
+Environment=http_proxy=<proxy>
+Environment=https_proxy=<proxy>
+Environment=no_proxy=localhost,127.0.0.1
ExecStart=/usr/lib/coreos/kubelet-wrapper \
```
I hope this might be of use to anyone that ends up here having the same problem as me.

**Comment by aaronlevy** [comment]
Seems that there a quite a few different issues being discussed here & I haven't been able to reproduce the original issue. I'm going to close this for now, but if there is still something unresolved, please reopen.

**Comment by saintx** [comment]
$ date
Mon Apr  4 17:14:40 EDT 2016
$ pwd
/Users/saintx/dev/coreos-kubernetes/single-node
$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	local-kube-config

nothing added to commit but untracked files present (use "git add" to track)
$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'coreos-alpha' is up to date...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 => 2222 (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: core
    default: SSH auth method: private key
    default: Warning: Connection timeout. Retrying...
==> default: Machine booted and ready!
==> default: Configuring and enabling network interfaces...
==> default: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> default: flag to force provisioning. Provisioners marked to run always will still run.
$ cat local-kube-config
kubectl config set-cluster vagrant-single-cluster --server=https://172.17.4.99:443 --certificate-authority=${PWD}/ssl/ca.pem
kubectl config set-credentials vagrant-single-admin --certificate-authority=${PWD}/ssl/ca.pem --client-key=${PWD}/ssl/admin-key.pem --client-certificate=${PWD}/ssl/admin.pem
kubectl config set-context vagrant-single --cluster=vagrant-single-cluster --user=vagrant-single-admin
kubectl config use-context vagrant-single
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"1", GitVersion:"v1.1.8", GitCommit:"a8af33dc07ee08defa2d503f81e7deea32dd1d3b", GitTreeState:"clean"}
$ . local-kube-config
cluster "vagrant-single-cluster" set.
user "vagrant-single-admin" set.
context "vagrant-single" set.
switched to context "vagrant-single".
$ kubectl get nodes
error: couldn't read version from server: Get https://172.17.4.99:443/api: dial tcp 172.17.4.99:443: i/o timeout

**Comment by aaronlevy** [comment]
In this case it's possible the api-server isn't actually running yet. The startup can take some time because it will need to download the kubelet container, and start all services / assets. 

If you give it a couple minutes does the api-server start responding? Or if not, is there anything of interest in the kubelet service log?

**Comment by xueshanf** [comment]
@aaronlevy I am still having the same issue. Hopefully I can help to figure out what's going on. I followed https://coreos.com/kubernetes/docs/latest/kubernetes-on-vagrant.html instructions for multi-node install.

I have systemctl status for flanned, kubelet and iptables -L, route -n and docker ps output below. Let me know if you need any other informations. Thanks!

```
$ vagrant status
Current machine states:

e1                        running (virtualbox)
c1                        running (virtualbox)
w1                        running (virtualbox)

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"1", GitVersion:"v1.1.8", GitCommit:"a8af33dc07ee08defa2d503f81e7deea32dd1d3b", GitTreeState:"clean"}
error: couldn't read version from server: Get https://172.17.4.101:443/api: dial tcp 172.17.4.101:443: connection refused
```

I can login to c1 and run the following:
```
core@c1 ~ $ systemctl status flanneld
● flanneld.service - Network fabric for containers
   Loaded: loaded (/usr/lib64/systemd/system/flanneld.service; disabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/flanneld.service.d
           └─40-ExecStartPre-symlink.conf.conf
   Active: active (running) since Tue 2016-04-05 05:34:34 UTC; 10h ago
     Docs: https://github.com/coreos/flannel
  Process: 1651 ExecStartPost=/usr/bin/docker run --net=host --rm --volume=/run:/run ${FLANNEL_IMG}:${FLANNEL_VER} /opt/bin/mk-docker-opts.sh -d /run/flannel_docker_opts.env -i (code=exited, status=0/SUCCESS)
  Process: 1527 ExecStartPre=/usr/bin/ln -sf /etc/flannel/options.env /run/flannel/options.env (code=exited, status=0/SUCCESS)
  Process: 1520 ExecStartPre=/usr/bin/touch ${FLANNEL_ENV_FILE} (code=exited, status=0/SUCCESS)
  Process: 1516 ExecStartPre=/usr/bin/mkdir -p ${ETCD_SSL_DIR} (code=exited, status=0/SUCCESS)
  Process: 1510 ExecStartPre=/usr/bin/mkdir -p /run/flannel (code=exited, status=0/SUCCESS)
  Process: 1505 ExecStartPre=/sbin/modprobe ip_tables (code=exited, status=0/SUCCESS)
 Main PID: 1531 (sdnotify-proxy)
   Memory: 9.2M
      CPU: 101ms
   CGroup: /system.slice/flanneld.service
           ├─1531 /usr/libexec/sdnotify-proxy /run/flannel/sd.sock /usr/bin/docker run --net=host --privileged=true --rm --volume=/run/flannel:/run/flannel --env=NOTIFY_SOCK...
           └─1539 /usr/bin/docker run --net=host --privileged=true --rm --volume=/run/flannel:/run/flannel --env=NOTIFY_SOCKET=/run/flannel/sd.sock --env=AWS_ACCESS_KEY_ID= ...

Apr 05 05:34:34 c1 sdnotify-proxy[1531]: I0405 05:34:34.797434 00001 ipmasq.go:50] Adding iptables rule: FLANNEL ! -d 224.0.0.0/4 -j MASQUERADE
Apr 05 05:34:34 c1 sdnotify-proxy[1531]: I0405 05:34:34.809381 00001 ipmasq.go:50] Adding iptables rule: POSTROUTING -s 10.2.0.0/16 -j FLANNEL
Apr 05 05:34:34 c1 sdnotify-proxy[1531]: I0405 05:34:34.811277 00001 ipmasq.go:50] Adding iptables rule: POSTROUTING ! -s 10.2.0.0/16 -d 10.2.0.0/16 -j MASQUERADE
Apr 05 05:34:34 c1 sdnotify-proxy[1531]: I0405 05:34:34.822878 00001 vxlan.go:153] Watching for L3 misses
Apr 05 05:34:34 c1 sdnotify-proxy[1531]: I0405 05:34:34.822878 00001 vxlan.go:159] Watching for new subnet leases
Apr 05 05:34:34 c1 sdnotify-proxy[1531]: I0405 05:34:34.825746 00001 vxlan.go:273] Handling initial subnet events
Apr 05 05:34:34 c1 sdnotify-proxy[1531]: I0405 05:34:34.825771 00001 device.go:159] calling GetL2List() dev.link.Index: 4
Apr 05 05:34:34 c1 systemd[1]: Started Network fabric for containers.
Apr 05 05:35:23 c1 sdnotify-proxy[1531]: I0405 05:35:23.448705 00001 vxlan.go:232] Subnet added: 10.2.47.0/24
```

```
core@c1 /etc/systemd/system $ systemctl status kubelet.service -l
● kubelet.service
   Loaded: loaded (/etc/systemd/system/kubelet.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2016-04-05 05:27:56 UTC; 10h ago
  Process: 1312 ExecStartPre=/usr/bin/mkdir -p /etc/kubernetes/manifests (code=exited, status=0/SUCCESS)
 Main PID: 1322 (kubelet)
   Memory: 8.7M
      CPU: 12.525s
   CGroup: /system.slice/kubelet.service
           ├─1322 /kubelet --api-servers=http://127.0.0.1:8080 --register-node=false --allow-privileged=true --config=/etc/kubernetes/manifests --hostname-override=172.17.4.101 --cluster_dns=10.3.0.10 --cluster_domain=cluster.local
           └─1737 journalctl -k -f

Apr 05 15:39:25 c1 kubelet-wrapper[1322]: E0405 15:39:25.402647    1322 kubelet.go:2291] Error updating node status, will retry: error getting node "172.17.4.101": Get http://127.0.0.1:8080/api/v1/nodes/172.17.4.101: dial tcp 127.0.0.1:8080: connection refused
Apr 05 15:39:25 c1 kubelet-wrapper[1322]: E0405 15:39:25.402830    1322 kubelet.go:2291] Error updating node status, will retry: error getting node "172.17.4.101": Get http://127.0.0.1:8080/api/v1/nodes/172.17.4.101: dial tcp 127.0.0.1:8080: connection refused
Apr 05 15:39:25 c1 kubelet-wrapper[1322]: E0405 15:39:25.403081    1322 kubelet.go:2291] Error updating node status, will retry: error getting node "172.17.4.101": Get http://127.0.0.1:8080/api/v1/nodes/172.17.4.101: dial tcp 127.0.0.1:8080: connection refused
```

```
$core@c1 /etc/systemd/system $ sudo  iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
           tcp  --  anywhere             anywhere             state NEW tcp dpt:https

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
DOCKER-ISOLATION  all  --  anywhere             anywhere
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain DOCKER (1 references)
target     prot opt source               destination

Chain DOCKER-ISOLATION (1 references)
target     prot opt source               destination
RETURN     all  --  anywhere             anywhere
```

```
core@c1 /etc/systemd/system $ route -n
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         10.0.2.2        0.0.0.0         UG    1024   0        0 eth0
10.0.2.0        0.0.0.0         255.255.255.0   U     0      0        0 eth0
10.0.2.2        0.0.0.0         255.255.255.255 UH    1024   0        0 eth0
10.2.0.0        0.0.0.0         255.255.0.0     U     0      0        0 flannel.1
10.2.68.0       0.0.0.0         255.255.255.0   U     0      0        0 docker0
172.17.4.0      0.0.0.0         255.255.255.0   U     0      0        0 eth1
```

```
core@c1 /etc/systemd/system $ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 08:00:27:fa:a9:e4 brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global dynamic eth0
       valid_lft 80283sec preferred_lft 80283sec
    inet6 fe80::a00:27ff:fefa:a9e4/64 scope link
       valid_lft forever preferred_lft forever
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 08:00:27:6c:5f:1f brd ff:ff:ff:ff:ff:ff
    inet 172.17.4.101/24 brd 172.17.4.255 scope global eth1
       valid_lft forever preferred_lft forever
    inet6 fe80::a00:27ff:fe6c:5f1f/64 scope link
       valid_lft forever preferred_lft forever
4: flannel.1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue state UNKNOWN group default
    link/ether 46:69:24:eb:58:ac brd ff:ff:ff:ff:ff:ff
    inet 10.2.68.0/16 scope global flannel.1
       valid_lft forever preferred_lft forever
    inet6 fe80::4469:24ff:feeb:58ac/64 scope link
       valid_lft forever preferred_lft forever
5: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default
    link/ether 02:42:12:27:d6:52 brd ff:ff:ff:ff:ff:ff
    inet 10.2.68.1/24 scope global docker0
       valid_lft forever preferred_lft forever
```

```
core@c1 /etc/systemd/system $ docker ps
CONTAINER ID        IMAGE                                  COMMAND             CREATED             STATUS              PORTS               NAMES
2a7662fa658b        gcr.io/google_containers/pause:0.8.0   "/pause"            10 hours ago        Up 10 hours                             k8s_POD.6d00e006_kube-proxy-172.17.4.101_kube-system_55cd7edd6b9f452ae2261be14a240e5b_68f2863d
9e07aa407878        gcr.io/google_containers/pause:0.8.0   "/pause"            10 hours ago        Up 10 hours                             k8s_POD.6d00e006_kube-podmaster-172.17.4.101_kube-system_7bd9ef00a6fcc78ab5409baa900f3e94_f3f9beab
00716cce8e9f        gcr.io/google_containers/pause:0.8.0   "/pause"            10 hours ago        Up 10 hours                             k8s_POD.6d00e006_kube-apiserver-172.17.4.101_kube-system_734dccbdca01b2f29191fceb5377dfb9_1f4fb81a
```

**Comment by aaronlevy** [comment]
So it doesn't look like any of the kubernetes control-plane is running (based on the docker ps output). Can you see if there are pods that had died (`docker ps -a`) and then their logs might give you more info (`docker logs <container id>`)

**Comment by xueshanf** [comment]
@aaronlevy I did docker ps -a, no died dockers. Out of desperation, I rebooted c1, and then it comes back with one error message which did not exist the first time:
```
MBPSF:vagrant sfeng$ vagrant ssh c1
Last login: Tue Apr  5 17:10:44 2016 from 10.0.2.2
CoreOS alpha (1000.0.0)
Update Strategy: No Reboots
Failed Units: 1
  coreos-cloudinit-336841772.service
```
```
core@c1 ~ $ systemctl status  coreos-cloudinit-336841772.service
● coreos-cloudinit-336841772.service - Unit generated and executed by coreos-cloudinit on behalf of user
   Loaded: loaded (/run/systemd/system/coreos-cloudinit-336841772.service; static; vendor preset: disabled)
  Drop-In: /run/systemd/system/coreos-cloudinit-336841772.service.d
           └─50-Description.conf, 50-ExecStart.conf
   Active: failed (Result: exit-code) since Tue 2016-04-05 15:50:02 UTC; 1h 22min ago
  Process: 737 ExecStart=/bin/bash /var/lib/coreos-cloudinit/scripts/336841772 (code=exited, status=1/FAILURE)
 Main PID: 737 (code=exited, status=1/FAILURE)

Apr 05 15:50:02 c1 systemd[1]: Started Unit generated and executed by coreos-cloudinit on behalf of user.
Apr 05 15:50:02 c1 bash[737]: Missing required config value: ETCD_ENDPOINTS
Apr 05 15:50:02 c1 systemd[1]: coreos-cloudinit-336841772.service: Main process exited, code=exited, status=1/FAILURE
Apr 05 15:50:02 c1 systemd[1]: coreos-cloudinit-336841772.service: Unit entered failed state.
Apr 05 15:50:02 c1 systemd[1]: coreos-cloudinit-336841772.service: Failed with result 'exit-code'.
```

The system does work now after the reboot. I am going to destroy everything to see if I can re-produce. 

```
core@c1 ~ $ docker ps
CONTAINER ID        IMAGE                                      COMMAND                  CREATED             STATUS              PORTS               NAMES
37ea5e760d13        quay.io/coreos/hyperkube:v1.1.8_coreos.0   "/hyperkube scheduler"   About an hour ago   Up About an hour                        k8s_kube-scheduler.d7af6bd8_kube-scheduler-172.17.4.101_kube-system_ba0ffcd76178cb64f5a40cb39237b663_84da12ba
8aed6eda3959        gcr.io/google_containers/pause:0.8.0       "/pause"                 About an hour ago   Up About an hour                        k8s_POD.6d00e006_kube-scheduler-172.17.4.101_kube-system_ba0ffcd76178cb64f5a40cb39237b663_52172f5e
d6bfe3c3578e        quay.io/coreos/hyperkube:v1.1.8_coreos.0   "/hyperkube controlle"   About an hour ago   Up About an hour                        k8s_kube-controller-manager.f55cdde6_kube-controller-manager-172.17.4.101_kube-system_e71b39ee832e3cbede008579a3b6e46f_73b3a8e4
e253a9610b47        gcr.io/google_containers/pause:0.8.0       "/pause"                 About an hour ago   Up About an hour                        k8s_POD.6d00e006_kube-controller-manager-172.17.4.101_kube-system_e71b39ee832e3cbede008579a3b6e46f_77cc512f
ed1277e4f024        quay.io/coreos/hyperkube:v1.1.8_coreos.0   "/hyperkube apiserver"   About an hour ago   Up About an hour                        k8s_kube-apiserver.fb534df3_kube-apiserver-172.17.4.101_kube-system_734dccbdca01b2f29191fceb5377dfb9_b59c2622
dd44ba6b3f04        quay.io/coreos/hyperkube:v1.1.8_coreos.0   "/hyperkube proxy --m"   About an hour ago   Up About an hour                        k8s_kube-proxy.924154b4_kube-proxy-172.17.4.101_kube-system_55cd7edd6b9f452ae2261be14a240e5b_e752dcdf
6ad4d665b798        gcr.io/google_containers/podmaster:1.1     "/podmaster --etcd-se"   About an hour ago   Up About an hour                        k8s_controller-manager-elector.2d387256_kube-podmaster-172.17.4.101_kube-system_7bd9ef00a6fcc78ab5409baa900f3e94_8dc77b91
5368fd0eb68d        gcr.io/google_containers/podmaster:1.1     "/podmaster --etcd-se"   About an hour ago   Up About an hour                        k8s_scheduler-elector.2611672a_kube-podmaster-172.17.4.101_kube-system_7bd9ef00a6fcc78ab5409baa900f3e94_80dc29ac
2cfb0500f7b1        gcr.io/google_containers/pause:0.8.0       "/pause"                 About an hour ago   Up About an hour                        k8s_POD.6d00e006_kube-apiserver-172.17.4.101_kube-system_734dccbdca01b2f29191fceb5377dfb9_81f13219
f90c0eddb02e        gcr.io/google_containers/pause:0.8.0       "/pause"                 About an hour ago   Up About an hour                        k8s_POD.6d00e006_kube-podmaster-172.17.4.101_kube-system_7bd9ef00a6fcc78ab5409baa900f3e94_c861042b
83585871ca35        gcr.io/google_containers/pause:0.8.0       "/pause"                 About an hour ago   Up About an hour                        k8s_POD.6d00e006_kube-proxy-172.17.4.101_kube-system_55cd7edd6b9f452ae2261be14a240e5b_70db16fc
```

**Comment by xueshanf** [comment]
@aaronlevy destroy and rebuild the cluster works! You are right that it does take some time. For me the kubelet-wrapper spent 8 minutes to download ACI image and only after that the controller-plane pods start showing up. It's about 10 minutes waiting. If you are on a slower network (like my earlier report, I was on slow home network), it would take much much longer.

**Comment by ReneSaenz** [comment]
I am still having this issue. Can someone please help?

$ kubectl --kubeconfig=kubeconfig get nodes
Unable to connect to the server: dial tcp 172.17.4.101:443: i/o timeout

CoreOS version

$ vagrant ssh c1
Last login: Wed Mar 15 04:38:29 UTC 2017 from 10.0.2.2 on pts/0
Container Linux by CoreOS alpha (1339.0.0)
Failed Units: 1
  update-engine.service



KUBECONFIG FILE

apiVersion: v1
clusters:
- cluster:
    certificate-authority: ssl/ca.pem
    server: https://172.17.4.101:443
  name: vagrant-multi-cluster
contexts:
- context:
    cluster: vagrant-multi-cluster
    user: vagrant-multi-admin
  name: vagrant-multi
current-context: vagrant-multi
kind: Config
preferences: {}
users:
- name: vagrant-multi-admin
  user:
    client-certificate: ssl/admin.pem
    client-key: ssl/admin-key.pem
